pkg/auth/azure.go

```diff
@@ -301,11 +257,14 @@ func (a *Azure) getBearerTokenOnBehalfOfUser(token string) (string, error) {
 		return "", err
 	}
 
-	var tokenResponse TokenResponse
-	if err := json.NewDecoder(response.Body).Decode(&tokenResponse); err != nil {
+	var groupsResponse AzureGroupsWithIDResponse
+	if err := json.NewDecoder(response.Body).Decode(&groupsResponse); err != nil {
 		return "", err
 	}
 
-	logrus.Debugf("Successfully retrieved on-behalf-of token")
-	return tokenResponse.AccessToken, nil
+	if len(groupsResponse.Groups) > 0 {
```

Comment:
shouldn't this be a == 1 check and not >0?

Reply:
I dont know if it is possible to have two groups with the same group email in AAD?
>0 works anyways so I choose it

Reply:
My concern was that in the unlikely event that two groups is registered with the same email one could possible grant admin access to the wrong group. But at least as far as I can see the email property is unique and then it does not matter, so I think this is good as is.